src/main/java/de/srendi/advancedperipherals/common/addons/dimstorage/DimChestIntegration.java

```diff
@@ -0,0 +1,55 @@
+package de.srendi.advancedperipherals.common.addons.dimstorage;
+
+import dan200.computercraft.api.lua.LuaFunction;
+import de.srendi.advancedperipherals.lib.peripherals.BlockEntityIntegrationPeripheral;
+import edivad.dimstorage.api.Frequency;
+import edivad.dimstorage.blockentities.BlockEntityDimChest;
+import net.minecraft.world.level.block.entity.BlockEntity;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.UUID;
+
+public class DimChestIntegration extends BlockEntityIntegrationPeripheral<BlockEntityDimChest> {
+
+    protected DimChestIntegration(BlockEntity entity) {
+        super(entity);
+    }
+
+    @NotNull
+    @Override
+    public String getType() {
+        return "dimChest";
+    }
+
+    @LuaFunction
```

Comment:
Please set `mainThread` to true to be thread safe with the world
`@LuaFunction(mainThread = true)`

Reply:
Fixed